statsd/statsd.go

```diff
@@ -159,9 +160,19 @@ type ClientInterface interface {
 	// Gauge measures the value of a metric at a particular time.
 	Gauge(name string, value float64, tags []string, rate float64) error
 
+	// Gauge measures the value of a metric at a given time.
+	// Even with client side aggregation enabled, there is no aggregation done on a metric
+	// sent using GaugeWithTimestamp: it is written as is in the serialization buffer.
```

Comment:
I this this should be more high level doc. Users reading the API doc for the client might not understand/know all the options for the client.

WDYT:

```suggestion
	// GaugeWithTimestamp measures the value of a metric at a given time.
	/// The value will bypass any aggregation on the client side and agent side. The Agent will simply forward this
	/// metrics to the backend. This is useful when sending points in the past.
```

Reply:
Using your suggestion, thanks.

Reply:
I just remove the " The Agent will simply forward this metrics to the Datadog." which is finally redundant with your previous sentence; if you don't mind.